astroscrappy/astroscrappy.pyx

```diff
@@ -176,6 +199,10 @@ def detect_cosmics(indat, inmask=None, float sigclip=4.5, float sigfrac=0.3,
     # Set the initial values to those of the data array
     cleanarr[:, :] = indat[:, :]
 
+    # Subtract the input sky model, if applicable.
+    if bkg is not None:
+        cleanarr -= bkg
+
```

Comment:
I think this should happen much later in the function. The background needs to be included when you set `cleanvar = cleanarr` at line 221.

Reply:
The background for the noise estimate gets restored again at L303, which needs to happen after the median filtering; see recent discussion at https://github.com/jehturner/astroscrappy/pull/1 for further explanation.